lib/uffizzi/cli/install.rb

```diff
@@ -0,0 +1,310 @@
+    HELM_DEPLOYED_STATUS = 'deployed'
+    CHART_NAME = 'uffizzi-app'
+    VALUES_FILE_NAME = 'helm_values.yaml'
+    DEFAULT_ISSUER = 'letsencrypt'
+    DEFAULT_NAMESPACE = 'uffizzi'
+    DEFAULT_APP_PREFIX = 'uffizzi'
+
+    desc 'application', 'Install uffizzi to cluster'
+    method_option :namespace, type: :string
+    method_option :domain, type: :string
+    method_option :'user-email', type: :string
+    method_option :'acme-email', type: :string
+    method_option :'user-password', type: :string
+    method_option :issuer, type: :string, enum: ['letsencrypt', 'zerossl']
+    method_option :'wildcard-cert-path', type: :string
+    method_option :'wildcard-key-path', type: :string
+    method_option :'without-wildcard-tls', type: :boolean
+    method_option :repo, type: :string
+    method_option :'print-values', type: :boolean
+    def application
+      run_installation do
+        if options.except(:repo, :'print-values').present?
+          validate_installation_options
+        else
+          ask_installation_params
+        end
+      end
+    end
+
+    desc 'wildcard-tls', 'Add wildcard tls from files'
+    method_option :domain, type: :string
+    method_option :cert, type: :string
+    method_option :key, type: :string
+    method_option :namespace, type: :string
+    method_option :repo, type: :string
+    def wildcard_tls
+      kubectl_exists?
+
+      params = if options.except(:repo).present? && wildcard_tls_options_valid?
+        {
+          namespace: options[:namespace] || DEFAULT_NAMESPACE,
+          domain: options[:domain],
+          wildcard_cert_path: options[:cert],
+          wildcard_key_path: options[:key],
+        }
+      else
+        namespace = Uffizzi.prompt.ask('Namespace: ', required: true, default: DEFAULT_NAMESPACE)
+        domain = Uffizzi.prompt.ask('Domain: ', required: true, default: 'example.com')
+        wildcard_cert_paths = ask_wildcard_cert(has_user_wildcard_cert: true, domain: domain)
+
+        { namespace: namespace, domain: domain }.merge(wildcard_cert_paths)
+      end
+
+      kubectl_add_wildcard_tls(params)
+      helm_values = helm_get_values(namespace, namespace)
+      helm_values['uffizzi-controller']['tlsPerDeploymentEnabled'] = false.to_s
+      create_helm_values_file(helm_values)
+      helm_set_repo unless options[:repo]
+      helm_install(release_name: namespace, namespace: namespace, repo: options[:repo])
+    end
+
+    default_task :application
+
+    private
+
+    def wildcard_tls_options_valid?
+      required_options = [:namespace, :domain, :cert, :key]
+      missing_options = required_options - options.symbolize_keys.keys
+
+      return true if missing_options.empty?
+
+      rendered_missing_options = missing_options.map { |o| "'--#{o}'" }.join(', ')
+
+      Uffizzi.ui.say_error_and_exit("No value provided for required options #{rendered_missing_options}")
+    end
+
+    def run_installation
+      kubectl_exists?
+      helm_exists?
```

Comment:
What if they don't exist? I don't see that the code execution is interrupted after this check.

Reply:
`execute_command` It takes care of that